src/ng/components/selection/selection.core.js

```diff
@@ -0,0 +1,84 @@
+import Directive from '../directive';
+import {VIEW_CORE_NAME, SELECTION_CORE_NAME} from '../../../definition';
+import Event from '../../../core/infrastructure/event';
+//import angular from 'angular';
+
+class SelectionCore extends Directive(SELECTION_CORE_NAME, {view: `^^${VIEW_CORE_NAME}`}) {
+	constructor($scope, $element, $attrs) {
+		super();
+
+		this.$scope = $scope;
+		this.$element = $element;
+		this.$attrs = $attrs;
+		
+		this.selectedItems = new Set();
```

Comment:
Yes, you're right, we have an array of selected rows within our model. We also don't have any kind of isSelected property. Therefore we can't just bind ng-checked to the row's field. We need to bind to kind of a function which determines if row selected or not. Generally this is bad practice and it might dramatically impact performance, but anyway I tried to make state(row) function as fastest as possible using Set lookup instead of naive direct search through the items array within our selection model.